argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,293 @@
+
+        with Pool(os.cpu_count()) as p:
+            accum = p.map(self.accumulate, gt_fpaths)
+
+        for frame_stats, frame_cls_to_inst in accum:
+            for cls_name, cls_stats in frame_stats.items():
+                data[cls_name].append(cls_stats)
+            for cls_name, num_inst in frame_cls_to_inst.items():
+                cls_to_ninst[cls_name] += num_inst
+
+        data = defaultdict(np.ndarray, {k: np.vstack(v) for k, v in data.items()})
+
+        columns = ["AP", "ATE", "ASE", "AOE", "CDS"]
+
+        # Initialize metrics table [0, 1.0, 1.0, 1.0, 0] represents the default value for each column.
+        init_data = {k: [0, 1.0, 1.0, 1.0, 0] for k in self.dt_cfg.detection_classes}
+        summary = pd.DataFrame.from_dict(init_data, orient="index", columns=columns)
+        summary_update = pd.DataFrame.from_dict(self.summarize(data, cls_to_ninst), orient="index", columns=columns)
+
+        summary.update(summary_update)
+        summary = summary.round(self.dt_cfg.significant_digits)
+        summary.index = summary.index.str.title()
+
+        summary.loc["Average Metrics"] = summary.mean().round(self.dt_cfg.significant_digits)
+        return summary
+
+    def accumulate(self, gt_fpath: Path) -> Tuple[DefaultDict[str, np.ndarray], DefaultDict[str, int]]:
+        """Accumulate the statistics for each LiDAR frame.
+
+        Args:
+            gt_fpath: Ground truth file path.
+
+        Returns:
+            cls_stats: Class statistics of shape (N, 8)
+            cls_to_ninst: Mapping of the class names to the number of instances in the ground
+                          truth dataset.
+        """
+        log_id = gt_fpath.parents[1].stem
+        logger.info(f"log_id = {log_id}")
+        ts = gt_fpath.stem.split("_")[-1]
+        dt_fpath = self.dt_fpath / f"{log_id}/per_sweep_annotations_amodal/" f"tracked_object_labels_{ts}.json"
+
+        dts = np.array(read_label(dt_fpath))
+        gts = np.array(read_label(gt_fpath))
+
+        cls_stats = defaultdict(list)
+        cls_to_ninst = defaultdict(int)
+        for dt_cls in self.dt_cfg.detection_classes:
+            dt_filtered = filter_instances(dts, dt_cls)
+            gt_filtered = filter_instances(gts, dt_cls)
+
+            logger.info(f"{dt_filtered.shape[0]} detections")
+            logger.info(f"{gt_filtered.shape[0]} ground truth")
+            if dt_filtered.shape[0] > 0:
+                cls_stats[dt_cls] = self.assign(dt_filtered, gt_filtered)
+
+            cls_to_ninst[dt_cls] = gt_filtered.shape[0]
+        return cls_stats, cls_to_ninst
+
+    def assign(self, dts: np.ndarray, gts: np.ndarray) -> np.ndarray:
+        """Attempt assignment of each detection to a ground truth label.
+
+        Args:
+            dts: Detections of shape (N,).
+            gts: Ground truth labels of shape (M,).
+
+        Returns:
+            True positives, false positives, scores, and translation errors.
```

Comment:
what is the shape and type of the return argument?

Reply:
let's resolve once we sync on return docstring style